exporters/dsexporter/dsexporter_test.go

```diff
@@ -2,45 +2,136 @@ package main
 
 import (
 	"testing"
+	"reflect"
+	"net/http"
+	"net/http/httptest"
 
-	"github.com/prometheus/client_golang/prometheus"
 	"github.com/stretchr/testify/assert"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 )
 
-func TestCustomCollector(t *testing.T) {
-	exporter := NewCustomCollector()
-	prometheus.MustRegister(exporter)
+func MockInclusterConfig() (*rest.Config) {
+    return nil
+}
+
+func MockNewForConfig(*rest.Config) (*kubernetes.Clientset) {
+	return nil
+}
+
+func TestNewKubeClient(t *testing.T) {
+	actualInclusterConfig := getInClusterConfig
+    getInClusterConfig = MockInclusterConfig
+    actualNewForConfig := getNewForConfig
+    getNewForConfig = MockNewForConfig
+
+    defer func() {
+        getInClusterConfig = actualInclusterConfig
+        getNewForConfig = actualNewForConfig
+    }()
```

Comment:
What's the reason for having this?

Reply:
it ensures original functions are restored and acts like a cleanup

Reply:
I think it makes sense for this function to take the client as input rather than creating it and then worrying about keeping the integrity of this global variable at the scope of this function.

Reply:
removed this test case since modified the code